### PR TITLE
HHH-13925 Upgrade to Gradle 6.3

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HHH-13925

~This is a preview: I ran the build locally to some extent, but I'd like a build from the CI. Also, I still need to run other builds, especially on JDK13 and 14, and to test publishing.~ => Done (see comments)

### CODE REVIEW

**Comment by yrodiere** [comment]
Everything seems in order:

* The CI build for JDK8 succeeded.
* The CI build for JDK11 failed on this PR, but that's a transient failure caused by problems transfering Maven artifacts. I relaunched the build.
* The CI build for JDK13 succeeded: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-orm-personal-yoann-jdk13/detail/hibernate-orm-personal-yoann-jdk13/6/pipeline
* The CI build for JDK14 succeeded: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-orm-personal-yoann-jdk14/detail/hibernate-orm-personal-yoann-jdk14/2/pipeline
* I managed to publish to my local Maven repository successfully
* I managed to publish artifacts to bintray successfully: https://bintray.com/yrodiere/hibernate-release-testing/hibernate-orm/999.4.0.Final
* I tested the hibernate-gradle-plugin with an application using Gradle 4.4: it still works, despite the plugin being compiled against Gradle 6 APIs.

File: gradlew [hunk 1]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
```
Comment (sebersole) [inline gradlew:3]: If we are going to add copyright headings it ought to be the correct one...

```
/*
 * Hibernate, Relational Persistence for Idiomatic Java
 *
 * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
 * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
 */
```

Comment (sebersole) [reply gradlew:3]: If you want to leave it as Apache, that's fine too.  I just mean the style

Comment (yrodiere) [reply gradlew:3]: I did not add these copyright headers, the Gradle Wrapper plugin did, automatically, when I upgraded. It's code from the Gradle Wrapper plugin, and I do not feel comfortable about changing either the license or the copyright holders.

Comment (yrodiere) [reply gradlew:3]: Besides, this file will be replaced automatically each time we upgrade to a newer Gradle version.

Comment (sebersole) [reply gradlew:3]: This is something new they have added then.  It is clearly new additions (new lines).  I just assumed you added them.  No problem then

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**